astropy/time/tests/test_custom_formats.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from astropy.time import Time, TimeFormat
+
+
+def test_custom_time_format_exception():
+    class SpecificException(ValueError):
+        pass
+
+    # Note there is no way to remove this after the test!
+    class Custom(TimeFormat):
+        name = "custom_time_format_test"
+
+        def set_jds(self, val, val2):
+            raise SpecificException
+
+    # with pytest.raises(SpecificException):
+    #    Time(7., format="custom_time_format_test")
+    try:
+        Time(7.0, format="custom_time_format_test")
+    except SpecificException:
```

Comment:
I think we'd want to be warned that this exception was leaking out, so let's not allow it.

But, it *would* be good to check that the `Custom` class raises it, with a direct test
```
def test_custom_direct(self):
    with pytest.raises(SpecificException):
        self.Custom(... fill in *all* arguments...)
```

Reply:
I'm actually not sure what it would mean to instantiate a TimeFormat object directly? Or how to do it exactly. I'll figure it out. Or, well, the __cause__ check should be sufficient, particularly combined with the very simple code of CustomFormat.